app/scripts/modules/openstack/common/selectField.directive.js

```diff
@@ -25,9 +25,13 @@ module.exports = angular.module('spinnaker.openstack.selectField.directive', [
         _.defaults(scope, {
           noOptionsMessage: '(No options available)',
           noSelectionMessage: '(No selection)',
-          labelColumnSize: 3
+          labelColumnSize: 3,
+          onChange: angular.noop
         });
 
+        //hack to get ng-model to work correctly with a bare scope variable
+        scope.s = scope;
```

Comment:
While this gets around the issue at hand, this is really dirty. I'd much rather see this directive rewritten as a component, moving the `link` logic into a controller, and using the `$onChanges` lifecycle hook in place of the `$watch`.

That said, if this is blocking you, I will merge it, as long as you add a TODO to refactor this to a component.